### ISSUE TITLE
HLSL: Trying to call Append on a TriangleStream argument causes compile error

### ISSUE THREAD

**Issue by tafuri** [KhronosGroup/glslang#1217]
Compiling the following valid HLSL shader

```hlsl
struct GSPS_INPUT
{
};

void EmitVertex(in GSPS_INPUT output, inout TriangleStream<GSPS_INPUT> TriStream)
{
    TriStream.Append( output );
}

[maxvertexcount(3)]
void main( triangle GSPS_INPUT input[3], inout TriangleStream<GSPS_INPUT> TriStream )
{
    EmitVertex(input[0], TriStream);
    EmitVertex(input[1], TriStream);
    EmitVertex(input[2], TriStream);
}
```
generates the following error:
```
ERROR: bug.gp:7: '' : unable to find output symbol for Append()
bug.gp(7): error at column 31, HLSL parsing failed.
```

**Comment by LoopDawg** [comment]
I suppose this is because of some parsing order issues, with the Append being seen before the definition of main which provides "triangle ...", in single pass compilation.  As a temporary workaround, you can try moving the EmitVertex function to after main, like this.

```
void EmitVertex(in GSPS_INPUT output, inout TriangleStream<GSPS_INPUT> TriStream);

[maxvertexcount(3)]
void main( triangle GSPS_INPUT input[3], inout TriangleStream<GSPS_INPUT> TriStream )
{
    EmitVertex(input[0], TriStream);
    EmitVertex(input[1], TriStream);
    EmitVertex(input[2], TriStream);
}

void EmitVertex(in GSPS_INPUT output, inout TriangleStream<GSPS_INPUT> TriStream)
{
    TriStream.Append( output );
}
```

**Comment by LoopDawg** [comment]
Ought to be fixed by #1378.

Sorry it took a while.
